driver/src/price_finding/naive_solver.rs

```diff
@@ -707,19 +694,19 @@ pub mod tests {
             return Ok(());
         }
 
-        if let Some(fee_token) = fee.as_ref().map(|fee| fee.token as usize) {
-            if solution.prices[fee_token] != BASE_PRICE {
+        if let Some(fee_token) = fee.as_ref().map(|fee| fee.token) {
+            if *solution.prices.get(&fee_token).unwrap_or(&BASE_PRICE) != BASE_PRICE {
```

Comment:
Shouldn't this fail if the solution doesn't specify the price of the fee token?

Reply:
Ya, it seems wrong to me as well, but don't forget that we wrote this line together.

Reply:
Sorry! I think it maybe should unwrap to 0 if there is no price.